sumologic/sumologic_role.go

```diff
@@ -53,6 +53,32 @@ func (s *Client) GetRole(id string) (*Role, error) {
 	return &role, nil
 }
 
+func (s *Client) GetRoleName(name string) (*Role, error) {
+	// TODO: check default limit count of 1000 and paginate
+	data, _, err := s.Get(fmt.Sprintf("v1/roles"))
+	if err != nil {
+		return nil, err
+	}
+
+	if data == nil {
+		return nil, fmt.Errorf("role with name '%s' does not exist", name)
+	}
+
+	var response RoleList
+	err = json.Unmarshal(data, &response)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range response.Roles {
+		if r.Name == name {
+			return &r, nil
+		}
+	}
```

Comment:
The api has a query param called `name`. We should use that instead of going through all the roles.

Reply:
I searched for it earlier, just like we have for collectors, but wasn't able to find it.
Let me change it, if we really have the support of `name` query parameter in roles API as well.

Reply:
![Screen Shot 2021-01-15 at 9 27 17 AM](https://user-images.githubusercontent.com/53231254/104758885-e8ffb780-5713-11eb-9f4f-1d93e919e035.png)
Should be the bottom one here

Reply:
Incorporated.